### PR TITLE
add detailed information in error logs

### PR DESCRIPTION
**Pull request by smallyard**
## What have you changed? (mandatory)

when I start up tikv-crl without any parameters, it exit without exactly reson. I don't know what happened until I  look at the source code. So, I pull this request.

## What are the type of the changes? (mandatory)

Improvement

## How has this PR been tested? (mandatory)

manual tests

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

no

## Does this PR affect tidb-ansible update? (mandatory)

no

### CODE REVIEW

File: src/bin/tikv-ctl.rs [hunk 1]
```diff
@@ -106,7 +106,7 @@ fn new_debug_executor(
             ))) as Box<DebugExecutor>
         }
         (Some(remote), None) => Box::new(new_debug_client(remote, mgr)) as Box<DebugExecutor>,
-        _ => unreachable!(),
+        _ => unreachable!("You have to set up the parameter --pb or --db!"),
```
Comment (Hoverbear) [inline src/bin/tikv-ctl.rs]: Maybe "Missing required parameters: --pb or --db" is a better message? It is not common for error messages to speak to you like their friend. :)

File: src/bin/tikv-ctl.rs [hunk 2]
```diff
@@ -106,7 +106,7 @@ fn new_debug_executor(
             ))) as Box<DebugExecutor>
         }
         (Some(remote), None) => Box::new(new_debug_client(remote, mgr)) as Box<DebugExecutor>,
-        _ => unreachable!(),
+        _ => unreachable!("Missing required parameters: --pd or --db"),
```
Comment (overvenus) [inline src/bin/tikv-ctl.rs]: `unreachable!` panics, what about `eprintln!()` + `process::exit()`?

Comment (Hoverbear) [reply src/bin/tikv-ctl.rs]: Good idea, let's make sure it exits with a non-zero code.

Comment (siddontang) [reply src/bin/tikv-ctl.rs]: why not using `perror_and_exit`?

Comment (smallyard) [reply src/bin/tikv-ctl.rs]: Is there need to create a custom exception?

Comment (BusyJay) [reply src/bin/tikv-ctl.rs]: I think `unreachable` or `panic` ensure that already.

**Comment by breeswish** [comment]
Seems that DCO is enforced now and we can't merge this PR @siddontang

**Comment by overvenus** [comment]
Hi @smallyard, could you sign off you commits? https://github.com/tikv/tikv/blob/master/CONTRIBUTING.md#signing-off-the-commit

**Comment by huachaohuang** [comment]
@smallyard feel free to reopen this if you are still interested, thanks.

**Pull request closed without merging**